src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs

```diff
@@ -0,0 +1,171 @@
+            string parentId = null;
+            if (requestMessage.GetHeader().ActivityIdParent != null)
+            {
+                // if W3C
+                Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+                Activity.ForceDefaultIdFormat = true;
+                parentId = requestMessage.GetHeader().ActivityIdParent;
+            }
+            else if (requestMessage.GetHeader().ActivityRequestId != null)
+            {
+                // if old scenario
+                parentId = requestMessage.GetHeader().ActivityRequestId;
+            }
+            else
+            {
+                // if activity not present
+                return;
+            }
+
+            // If there are any Listeners then create Activity using ActivitySource
+            var activity = activitySource.StartActivity("StatefulDatabaseIncomingRemoteCall", ActivityKind.Server, CreateActivityContextFromTraceParent(parentId));
+            bool activitySourceListenerPresent = true;
+
+            if (activity == null)
+            {
+                activitySourceListenerPresent = false;
+                activity = new Activity("StatefulDatabaseIncomingRemoteCall");
+                activity.SetParentId(parentId);
+            }
+
+            if (!string.IsNullOrEmpty(parentId))
+            {
+                // if W3C
+                if (requestMessage.GetHeader().ActivityIdTraceStateHeader != null)
+                {
+                    activity.TraceStateString = requestMessage.GetHeader().ActivityIdTraceStateHeader;
+                }
+            }
+
+            // We expect baggage to be empty by default
+            // Only very advanced users will be using it in near future, we encourage them to keep baggage small (few items)
+            if (requestMessage.GetHeader().ActivityIdBaggage != null)
+            {
+                var baggage = requestMessage.GetHeader().ActivityIdBaggage;
+                foreach (var item in baggage)
+                {
+                    activity.AddBaggage(item.Key, item.Value);
+                }
+            }
+
+            if (!activitySourceListenerPresent)
+            {
+                activity.Start();
+            }
+        }
+
+        /// <summary>
+        /// Determines whether this instance is present.
+        /// </summary>
+        /// <returns><c>true</c> if this instance is present; otherwise, <c>false</c>.</returns>
+        internal static bool IsPresent()
+        {
+            return Activity.Current != null;
+        }
+
+        /// <summary>
+        /// Tries the get.
+        /// </summary>
+        /// <returns><c>true</c> if XXXX, <c>false</c> otherwise.</returns>
+        internal static Activity Get()
+        {
+            return Activity.Current;
+        }
+
+        /// <summary>
+        /// Sets the specified activity.
+        /// </summary>
+        /// <param name="activity">The activity.</param>
+        internal static void Set(Activity activity)

```

Comment:
Where are Set and Clear methods used?

Reply:
Not used now, I was thinking if it would be used in the future. Removing them for now, will add later if needed for some other feature/implementation/scenario